### PR TITLE
[Android] Geo search API dev preview

### PR DESCRIPTION
**Pull request by raphkim**
_Description of changes:_

Add documentation for Android Geo m1 release 2 (Search API). 

Depends on:
- [x] Wait for [code change to be merged](https://github.com/aws-amplify/amplify-android/pull/1558).
- [x] Wait for Amplify Android release.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/fragments/lib/geo/android/existing-resources.mdx [hunk 1]
```diff
@@ -10,15 +10,21 @@ In order to manually configure Amplify Geo category, you must edit `amplifyconfi
     "geo": {
         "plugins": {
             "awsLocationGeoPlugin": {
+              "region": <YOUR_REGION>,
```
Comment (changxu0306) [inline src/fragments/lib/geo/android/existing-resources.mdx:13]:
```suggestion
                "region": <YOUR_REGION>,
```

File: src/fragments/lib/geo/android/search/10_search_by_text.mdx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        for (final Place place : result.getPlaces()) {
+            Log.i("MyAmplifyApp", place.toString());
+        }
+    },
+    error -> Log.e("MyAmplifyApp", "Failed to search for " + searchQuery, error)
+);
+```
+
+</Block>
+<Block name="Kotlin">
```
Comment (changxu0306) [inline src/fragments/lib/geo/android/search/10_search_by_text.mdx:21]: Is Kotlin Coroutine and RxJava supported?

File: src/fragments/lib/geo/android/search/10_search_by_text.mdx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+```kotlin
+val searchQuery = "Amazon Go"
+Amplify.Geo.searchByText(searchQuery,
+    {
+        for (place in it.places) {
+            Log.i("MyAmplifyApp", place.toString())
+        }
+    },
+    { Log.e("MyAmplifyApp", "Failed to get search for $searchQuery", it) }
```
Comment (changxu0306) [inline src/fragments/lib/geo/android/search/10_search_by_text.mdx:31]: To make it consistent with Java example: 
```suggestion
    { Log.e("MyAmplifyApp", "Failed to search for $searchQuery", it) }
```

File: src/fragments/lib/geo/android/search/10_search_by_text.mdx (hunk 2 above)
Comment (raphkim) [reply src/fragments/lib/geo/android/search/10_search_by_text.mdx:21]: not yet

**Review by changxu0306 — APPROVED** [review]

**Review by gpanshu — APPROVED** [review]

**Pull request merged**